File: backend/models/Expense.js
// models/Expense.js
import mongoose from 'mongoose';

const ExpenseSchema = new mongoose.Schema({
  user_id: { type: mongoose.Schema.Types.ObjectId, required: true, ref: 'User' },
  amount: { type: Number, required: true },
  category: { type: String, required: true }, // e.g., Food, Transport, Entertainment
  description: { type: String },
  date: { type: Date, default: Date.now }
});

export default mongoose.model('Expense', ExpenseSchema);

File: frontend/expensetracker/lib/models/expense.dart
class Expense {
  final String id;
  final String userId;
  final double amount;
  final String category;
  final String description;
  final String date;

  Expense({
    required this.id,
    required this.userId,
    required this.amount,
    required this.category,
    required this.description,
    required this.date,
  });

  factory Expense.fromJson(Map<String, dynamic> json) {
    return Expense(
      id: json['_id'],
      userId: json['user_id'],
      amount: json['amount'].toDouble(),
      category: json['category'],
      description: json['description'],
      date: json['date'],
    );
  }
}
